modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/CustomViewsControls.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm, {ClayInput} from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import {openModal} from 'frontend-js-web';
+import React, {useContext, useState} from 'react';
+
+import FrontendDataSetContext from './../../FrontendDataSetContext';
+
+const CustomViewsControls = () => {
+	const [viewsDropdownActive, setViewsDropdownActive] = useState(false);
+	const [actionsDropdownActive, setActionsDropdownActive] = useState(false);
+
+	const {namespace} = useContext(FrontendDataSetContext);
+
+	const SaveCustomViewModalBody = () => {
```

Comment:
Can we define this component outside the main closure? 🥺

Reply:
See https://github.com/liferay-frontend/liferay-portal/pull/2509#discussion_r918793311